src/components/Input.js

```diff
@@ -159,13 +159,13 @@ class Input extends Component {
     if (typeof this.props.onBlur === "function") this.props.onBlur(target.value);
   };
 
-  _onChange = ({ target }) => {
-    let value;
-    typeof this.props.onChange === "function"
-      ? (value = this.props.onChange(target.value))
-      : (value = target.value);
+  _onChange = ({ target }, onChange) => {
+    let value = target.value;
     if (this.props.maxLength !== 9999 && value.length > this.props.maxLength) return false;
+    if (onChange) value = onChange(target.value);
```

Comment:
I would check `typeof onChange === function` here to prevent accidental triggering of it as a string or something.